lms/djangoapps/grades/tests/test_new.py

```diff
@@ -239,56 +240,155 @@ def test_save_and_load(self):
 
 
 @ddt.ddt
-class TestMultipleProblemTypesSubsectionScores(ModuleStoreTestCase, ProblemSubmissionTestMixin):
+class TestMultipleProblemTypesSubsectionScores(SharedModuleStoreTestCase):
     """
     Test grading of different problem types.
     """
 
-    default_problem_metadata = {
-        u'graded': True,
-        u'weight': 2.5,
-        u'max_score': 7.0,
-        u'due': datetime.datetime(2099, 3, 15, 12, 30, 0, tzinfo=pytz.utc),
-    }
-
-    COURSE_NAME = u'Problem Type Test Course'
-    COURSE_NUM = u'probtype'
+    SCORED_BLOCK_COUNT = 6
+    ACTUAL_TOTAL_POSSIBLE = 16.0
 
     def setUp(self):
         super(TestMultipleProblemTypesSubsectionScores, self).setUp()
         password = u'test'
         self.student = UserFactory.create(is_staff=False, username=u'test_student', password=password)
         self.client.login(username=self.student.username, password=password)
         self.request = get_request_for_user(self.student)
-        self.course = CourseFactory.create(
-            display_name=self.COURSE_NAME,
-            number=self.COURSE_NUM
+        self.course_structure = get_course_blocks(self.student, self.course.location)
+
+    @classmethod
+    def load_scoreable_course(cls):
+        """
+        This test course lives at common/test/data/scoreable.
+
+        It contains six scoreable blocks in one sequence:
+
+        - CAPA
+        - ORA
+        - SGA (Staff Graded Assignment)
+        - LTI
+        - LTI Consumer
+        - Randomized Content Block (containing Library Content with CAPA
+          problems)
+
+        To expand coverage to other block types, you can either edit the course
+        xml directly, or do the following:
+
+        1. Zip up the scoreable directory into a tarball:
+
+            tar cvzf course.tar.gz common/test/data/scoreable'
+
+        2. Import the tarball into studio.
+        3. Add the new blocks.
+        4. Export the modified course.
+        5. Unzip to a temporary directory:
+
+            cd /tmp
+            tar xvzf ~/Downloads/course.*.tar.gz
+            somewhere, and
+
+        6. Copy the data back into the test directory:
+
+            rsync -avz --delete-after /tmp/course/ /path/to/edx-platform/common/test/data/scoreable
+
+        You will need to update SCORED_BLOCK_COUNT and ACTUAL_TOTAL_POSSIBLE
+        above to reflect the real number of scoreable blocks in the course.
+        """
+        course_items = import_course_from_xml(
+            cls.store,
+            'test_user',
+            TEST_DATA_DIR,
+            source_dirs=['scoreable'],
+            static_content_store=None,
+            target_id=cls.store.make_course_key('edX', 'scoreable', '3000'),
+            raise_on_failure=True,
+            create_if_not_present=True,
+        )
+
+        cls.course = course_items[0]
+
+    @classmethod
+    def setUpClass(cls):
```

Comment:
Nit: Can `setUpClass` be defined at the top (above `setUp`)?  Otherwise, as a first-time read of this class, it was unclear where the `self.course` in `setUp` came from.